src/main/scala/swiss/dasch/api/AuthService.scala

```diff
@@ -59,14 +71,26 @@ final case class AuthServiceLive(jwtConfig: JwtConfig) extends AuthService {
     val subVal = if (claim.subject.isDefined) { Validation.succeed(claim) }
     else { Validation.fail(AuthenticationError.subjectMissing()) }
 
+    val authScope =
+      Validation
+        .fromEither(
+          for {
+            contents  <- Right(Some(claim.content).filter(_.nonEmpty))
+            parsed    <- contents.map(_.fromJson[JwtContents]).getOrElse(Right(JwtContents()))
+            authScope <- parsed.scope.map(AuthScope.parse).getOrElse(Right(AuthScope.Empty))
+          } yield authScope,
+        )
+        .mapError(AuthenticationError.invalidContents)
+
     ZIO.fromEither(
       Validation
-        .validateWith(issVal, audVal, subVal)(
+        .validateWith(authScope, issVal, audVal, subVal)(
           (
+            authScope,
             _,
             _,
             _,
-          ) => claim,
+          ) => (authScope, claim),
         )
         .toEither,
```

Comment:
suggestion (non-blocking; not-introduced-by-you): As far as I recall, Validation has a `.toZio` method, that could spare us the `toEither` and `ZIO.fromEither` here

Reply:
Great catch!